stable_baselines3/her/her.py

```diff
@@ -0,0 +1,495 @@
+
+    @classmethod
+    def load(cls, load_path: str, env: Optional[GymEnv] = None, **kwargs) -> "BaseAlgorithm":
+        """
+        Load the model from a zip-file
+
+        :param load_path: the location of the saved data
+        :param env: the new environment to run the loaded model on
+            (can be None if you only need prediction from a trained model) has priority over any saved environment
+        :param kwargs: extra arguments to change the model when loading
+        """
+        data, params, tensors = load_from_zip_file(load_path)
+
+        if "policy_kwargs" in data:
+            for arg_to_remove in ["device"]:
+                if arg_to_remove in data["policy_kwargs"]:
+                    del data["policy_kwargs"][arg_to_remove]
+
+        if "policy_kwargs" in kwargs and kwargs["policy_kwargs"] != data["policy_kwargs"]:
+            raise ValueError(
+                f"The specified policy kwargs do not equal the stored policy kwargs."
+                f"Stored kwargs: {data['policy_kwargs']}, specified kwargs: {kwargs['policy_kwargs']}"
+            )
+
+        # check if observation space and action space are part of the saved parameters
+        if "observation_space" not in data or "action_space" not in data:
+            raise KeyError("The observation_space and action_space were not given, can't verify new environments")
+        # check if given env is valid
+        if env is not None:
+            # check if wrapper for dict support is needed
+            if isinstance(env.observation_space, gym.spaces.dict.Dict):
+                env = ObsDictWrapper(env)
+            check_for_correct_spaces(env, data["observation_space"], data["action_space"])
+        # if no new env was given use stored env if possible
+        if env is None and "env" in data:
+            env = data["env"]
+
+        kwargs = {}
+        if "use_sde" in data and data["use_sde"]:
+            kwargs["use_sde"] = True
+
+        # noinspection PyArgumentList
+        her_model = cls(
+            policy=data["policy_class"],
+            env=env,
+            model_class=data["model_class"],
+            n_sampled_goal=data["n_sampled_goal"],
+            goal_selection_strategy=data["goal_selection_strategy"],
+            online_sampling=data["online_sampling"],
+            learning_rate=data["learning_rate"],
+            max_episode_length=data["max_episode_length"],
+            policy_kwargs=data["policy_kwargs"],
+            _init_setup_model=True,  # pytype: disable=not-instantiable,wrong-keyword-args
+            **kwargs,
+        )
+
+        # load parameters
+        her_model.model.__dict__.update(data)
+        her_model.__dict__.update(kwargs)
+
+        her_model._total_timesteps = her_model.model._total_timesteps
+        her_model.num_timesteps = her_model.model.num_timesteps
+        her_model._episode_num = her_model.model._episode_num
+
+        # put state_dicts back in place
```

Comment:
you may need to update that part (it was improved recently: https://github.com/DLR-RM/stable-baselines3/pull/138)